run.go

```diff
@@ -96,12 +96,18 @@ func (s *Simulation) AdvanceFrame() error {
 
 	if len(s.queue) > 0 {
 		//check if the current action is executable right now; if not then delay
-		delay := s.C.AnimationCancelDelay(s.queue[0].Typ)
+		actionType := s.queue[0].Typ
+		delay := s.C.AnimationCancelDelay(actionType)
 		if delay > 0 {
 			s.skip = delay
 			return nil
 		}
 
+		if s.C.Flags.EnergyCalcMode && actionType == core.ActionBurst {
```

Comment:
Instead of modifying the simulation run function here, it's better to hook on to burst events. Something like the following code would work:

```go
		s.C.Events.Subscribe(core.PreBurst, func(args ...interface{}) bool {
			activeChar := s.C.Chars[s.C.ActiveChar]
			//log your stats here
		}, "reaction-log")
```

It's preferable not to modify the main code here and instead rely on adding in subscription to events if energy mode is enabled.

Reply:
Resolved